src/plugins/dashboard/public/dashboard_actions/library_notification_action.test.tsx

```diff
@@ -6,91 +6,57 @@
+    action = new LibraryNotificationAction(unlinkAction);
+    context = {
+      embeddable: {
+        viewMode: viewModeSubject,
+        canUnlinkFromLibrary: jest.fn().mockResolvedValue(true),
+        unlinkFromLibrary: jest.fn(),
+      },
+    };
   });
 
-  if (isErrorEmbeddable(contactCardEmbeddable)) {
-    throw new Error('Failed to create embeddable');
-  }
-  embeddable = embeddablePluginMock.mockRefOrValEmbeddable<
-    ContactCardEmbeddable,
-    ContactCardEmbeddableInput
-  >(contactCardEmbeddable, {
-    mockedByReferenceInput: { savedObjectId: 'testSavedObjectId', id: contactCardEmbeddable.id },
-    mockedByValueInput: { firstName: 'Kibanana', id: contactCardEmbeddable.id },
+  it('is compatible when api meets all conditions', async () => {
+    expect(await action.isCompatible(context)).toBe(true);
   });
-  embeddable.updateInput({ viewMode: ViewMode.EDIT });
-});
 
-test('Notification is incompatible with Error Embeddables', async () => {
-  const action = new LibraryNotificationAction(unlinkAction);
-  const errorEmbeddable = new ErrorEmbeddable(
-    'Wow what an awful error',
-    { id: ' 404' },
-    embeddable.getRoot() as IContainer
-  );
-  expect(await action.isCompatible({ embeddable: errorEmbeddable })).toBe(false);
-});
+  it('is incompatible when api is missing required functions', async () => {
+    const emptyContext = { embeddable: {} };
+    expect(await action.isCompatible(emptyContext)).toBe(false);
+  });
 
-test('Notification is shown when embeddable on dashboard has reference type input', async () => {
-  const action = new LibraryNotificationAction(unlinkAction);
-  embeddable.updateInput(await embeddable.getInputAsRefType());
-  expect(await action.isCompatible({ embeddable })).toBe(true);
-});
+  it('is incompatible when can unlink from library resolves to false', async () => {
+    context.embeddable.canUnlinkFromLibrary = jest.fn().mockResolvedValue(false);
+    expect(await action.isCompatible(context)).toBe(false);
+  });
 
-test('Notification is not shown when embeddable input is by value', async () => {
-  const action = new LibraryNotificationAction(unlinkAction);
-  embeddable.updateInput(await embeddable.getInputAsValueType());
-  expect(await action.isCompatible({ embeddable })).toBe(false);
-});
+  it('calls onChange when view mode changes', async () => {
+    const onChange = jest.fn();
+    updateViewMode('view');
+    action.subscribeToCompatibilityChanges(context, onChange);
+    await waitFor(() => expect(onChange).toHaveBeenCalledWith(false, action));
+  });
 
-test('Notification is not shown when view mode is set to view', async () => {
-  const action = new LibraryNotificationAction(unlinkAction);
-  embeddable.updateInput(await embeddable.getInputAsRefType());
-  embeddable.updateInput({ viewMode: ViewMode.VIEW });
-  expect(await action.isCompatible({ embeddable })).toBe(false);
+  it('is incompatible when api publishes a saved object id', async () => {});
```

Comment:
Good call out - I've removed this test because it is a leftover from when I was checking specifically for saved object IDs, rather than letting the API itself determine what `unlink` means.

This is covered by the test above `'is incompatible when can unlink from library resolves to false'`